test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
+
+        # Check that exec_vnode is a KNL vnode.
+        self.server.status(JOB, 'exec_vnode', id=job_id)
+        evnode = job.execvnode()[0].keys()[0]
+        if evnode in klist:
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
+        # Delete the Job.
+        self.server.delete(job_id, wait=True)
+
+        # Unset provisioning attributes.
+        self.unset_provisioning()
+
+    def test_job_request_subchunk(self):
+        """
+        Test job request consist of subchunks with and without aoe resource.
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='Test is NA for craysim')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set provisioning attributes.
+        self.set_provisioning()
+
+        # Submit job that request sub-chunk with and without aoe resources
+        req_aoe = self.request_current_aoe()
+        job = Job(TEST_USER)
+
+        job.create_script(
+                "#PBS -joe -o localhost:/tmp -lselect=1:ncpus=1:aoe=%s+1:ncpus=12\n" % req_aoe +
+                "cd /tmp\n" +
+                "aprun -B sleep 10\n" +
+                "sleep 10")
+
+        job_id = self.server.submit(job)
+        self.server.expect(JOB, {'job_state': 'R'}, id=job_id)
+
+        # Check that exec_vnode is a KNL vnode.
+        self.server.status(JOB, 'exec_vnode', id=job_id)
+        evnode = job.get_vnodes()
+        if evnode[0] in klist:
+            self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+        if evnode[1] not in klist:
+            self.logger.info("exec_vnode %s is a NON KNL vnode." % (evnode))
+            rv = 1
+        else:
+            self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
+            rv = 0
+        self.assertTrue(rv == 1)
+
+        # Delete the Job.
+        self.server.delete(job_id, wait=True)
+
+        # Unset provisioning attributes.
+        self.unset_provisioning()
+
+    def test_pbs_alps_in_sync(self):
+        """
+        Check for the presence of message indicating PBS and ALPS are
+        in sync.
+        """
+        # Determine if BASIL 1.7 is supported.
+        rv = self.mom.log_match(
+                "This Cray system supports the BASIL 1.7 protocol.", n='ALL')
+        if not rv:
+            self.skip_test('Skipping this test because there are no KNL nodes present.')
+            return
```

Comment:
You won't skip your test because log_match() will throw an exception and error your test.  Put a try/except around the log_match() so you can catch the log match failure.  The exception type is PtlLogMatchError